src/connection/Connection.ts

```diff
@@ -152,8 +151,7 @@ export class Connection {
      * but it also can setup a connection pool with database to use.
      */
     async connect(): Promise<this> {
-        if (this.isConnected)
-            throw new CannotConnectAlreadyConnectedError(this.name);
+        if (this.isConnected) return this;
```

Comment:
hey, you cannot simply take and change exist logic this way. This change is out of scope of what you are doing, please remove it.